wordpress-mobile/WordPress-iOS
# Add new Post/Site Preview UI for #13197

Implements a new `PreviewWebKitViewController` subclass which handles the preview specific logic for the UI described in #13197. The preview toolbar item and functionality will be added in a followup PR.

## How to Test

- Open any Post or Site Page preview from the Editor screen, or the Post/Pages lists.
- Ensure the following:
    - Post pages open any link non-authentication link in Safari
    - Site pages open any link on the same host in the Web View but all other non-authentication links in Safari
    - Toolbar displays the appropriate items as listed below in **Behavior** or in **Notes** from the [original design](https://wp.me/paCL20-aw).

## Behavior

Toolbar item logic:
- For un-published posts (drafts, scheduled, etc.): `Publish`, `Preview (coming)`
- For published posts: `Share`, `Safari`
- For Site Pages: `Back`, `Forward`, `Share`, `Safari`

The shown toolbar items match the behavior of the WebView when it receives a navigation action.

- For un-published and published posts, tapping on any link opens Safari
- For site pages all links on the same host are opened in the WebView, all other links open Safari

## Code

`PreviewWebKitViewController` is a subclass of `WebKitViewController` which provides most of the buttons and basic navigation handling.

The `LinkBehavior` type has been added to handle navigation actions depending on which type of preview has been opened.

## Screenshots

### Published Post

<details>
  <summary>
Screenshot
  </summary>

![Simulator Screen Shot - iPhone 11 Pro Max - 2020-01-13 at 10 54 18](https://user-images.githubusercontent.com/3250/72279542-9583f680-35f3-11ea-8704-03fc426c82e3.png)
</details>

### Draft Post

<details>
  <summary>
Screenshot
  </summary>

![Simulator Screen Shot - iPhone 11 Pro Max - 2020-01-13 at 10 54 55](https://user-images.githubusercontent.com/3250/72279536-93ba3300-35f3-11ea-820b-73e96ed7b722.png)
</details>

### Site Page

<details>
  <summary>
Screenshot
  </summary>

![Simulator Screen Shot - iPhone 11 Pro Max - 2020-01-13 at 10 55 23](https://user-images.githubusercontent.com/3250/72279529-8f8e1580-35f3-11ea-8300-2aef7d2a6665.png)
</details>

PR submission checklist:


- [x] ~I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.~ Added once Feature Flag is enabled
